cibuildwheel/macos.py

```diff
@@ -48,37 +50,64 @@ def call(args, env=None, cwd=None, shell=False):
+        python_package_identifier = 'org.python.Python.PythonFramework-{}'.format(version)
         if python_package_identifier not in installed_system_packages:
             # download the pkg
-            call(['curl', '-L', '-o', '/tmp/Python.pkg', config.url])
+            call(['curl', '-L', '-o', '/tmp/Python.pkg', url])
             # install
             call(['sudo', 'installer', '-pkg', '/tmp/Python.pkg', '-target', '/'])
             # patch open ssl
-            if config.version == '3.5':
+            if version == '3.5':
                 call(['curl', '-fsSLo', '/tmp/python-patch.tar.gz', 'https://github.com/mayeut/patch-macos-python-openssl/releases/download/v1.0.2t/patch-macos-python-%s-openssl-v1.0.2t.tar.gz' % config.version])
-                call(['sudo', 'tar', '-C', '/Library/Frameworks/Python.framework/Versions/%s/' % config.version, '-xmf', '/tmp/python-patch.tar.gz'])
+                call(['sudo', 'tar', '-C', '/Library/Frameworks/Python.framework/Versions/{}/'.format(version), '-xmf', '/tmp/python-patch.tar.gz'])
+        return '/Library/Frameworks/Python.framework/Versions/{}/bin'.format(version)
+    
+    def install_pypy(version, url):
+        pypy_tar_bz2 = url.rsplit('/', 1)[-1]
+        assert pypy_tar_bz2.endswith(".tar.bz2")
+        pypy_base_filename = os.path.splitext(os.path.splitext(pypy_tar_bz2)[0])[0]
+        installation_path = os.path.join('/tmp', pypy_base_filename)
+        if not os.path.exists(installation_path):
+            call(['curl', '-L', '-o', os.path.join("/tmp", pypy_tar_bz2), url])
+            call(['tar', '-C', '/tmp', '-xf', os.path.join("/tmp", pypy_tar_bz2)])
+        return os.path.join(installation_path, 'bin')
+
+
+    # get latest pip once and for all
+    call(['curl', '-L', '--retry', '3', '--retry-delay', '3', '-o', get_pip_script, get_pip_url])
+
+    for config in python_configurations:
+        if config.identifier.startswith('cp'):
+            installation_bin_path = install_cpython(config.version, config.url)
+            python_executable = 'python3' if config.version[0] == '3' else 'python'
+            pip_executable = 'pip3' if config.version[0] == '3' else 'pip'
+        elif config.identifier.startswith('pp'):
+            installation_bin_path = install_pypy(config.version, config.url)
+            python_executable = 'pypy3' if config.version[0] == '3' else 'pypy'
+            pip_executable = 'pip3' if config.version[0] == '3' else 'pip'
+            # this four lines apply patch for fix pypy bug with platform tag on macos
+            if config.version[0] == '3':
+                patch_file = os.path.abspath(os.path.join(os.path.dirname(__file__), 'resources', 'pypy3.6.patch'))
+                sysconfigdata_file = os.path.join(os.path.dirname(installation_bin_path), 'lib_pypy', '_sysconfigdata.py')
+                subprocess.check_call(['patch', sysconfigdata_file, patch_file , '-N'])
```

Comment:
There is a reason that I use `subprocess.run` instead `subprocess.check_call`. Even with option `-N` `patch` return nonzeros status if patch is already applied...

Reply:
Argh, sorry. But so, Python 2 doesn't know `subprocess.run` :-|

Reply:
Changing it to `call`, then?

Reply:
Change to `subrpocess.call`? It is still avaliable.